### DISCUSSION TITLE
fire componentWillDisappear event before starting transition animation

### DISCUSSION THREAD

**Discussion by flexsurfer** [wix/react-native-navigation#7362] [Enhancements & New Ideas]
## 🚀 Feature

it would be useful to know in JS when transition animation is started so some computation may be done in that time, and the previous screen will be opened updated 

# Implementation 

fire `componentWillDisappear` before starting transition animation

EDITED:
found the comment https://github.com/wix/react-native-navigation/issues/4793#issuecomment-467748525

but I think these events still will be useful

**Comment by erennyuksell** [comment]
i need this definitely
